azure.go

```diff
@@ -120,94 +141,94 @@ func (p *azureProvider) Verifier() tokenVerifier {
 }
 
 func (p *azureProvider) ComputeClient(subscriptionID string) (computeClient, error) {
-	authorizer, err := p.getAuthorizer()
+	cred, err := p.getTokenCredential()
+	if err != nil {
+		return nil, err
+	}
+
+	clientOptions := &arm.ClientOptions{
+		ClientOptions: policy.ClientOptions{
+			Transport: transporter{
+				pluginEnv: p.settings.PluginEnv,
+				sender:    p.httpClient,
+			},
+		},
+	}
```

Comment:
It looks like the `cloud.Configuration` is the replacement for what was [`azure.Environment`](https://github.com/hashicorp/vault-plugin-auth-azure/blob/main/azure.go#L210)? I think we'll want to continue to set the `cloud.Configuration` as we did for `azure.Environment` before when constructing eash of these clients.

I think it would look like:
```suggestion
	clientOptions := &arm.ClientOptions{
		ClientOptions: policy.ClientOptions{
		        Cloud: p.settings.CloudConfig,
			Transport: transporter{
				pluginEnv: p.settings.PluginEnv,
				sender:    p.httpClient,
			},
		},
	}
```

Reply:
Ah nice find! Thanks for catching this, fixed in latest commit